### ISSUE TITLE
rspec-rails under `:development, :test` groups results in double executing all specs!

### ISSUE THREAD

**Issue by jmuheim** [rspec/rspec-rails#904]
In your README you write:

> Add rspec-rails to both the :development and :test groups in the Gemfile

But this results in my specs are executed twice when running `rake spec`:

```
rake spec
[deprecated] I18n.enforce_available_locales will default to true in the future. If you really want to skip validation of your locale you can set I18n.enforce_available_locales = false to avoid this message.
/Users/josh/.rvm/rubies/ruby-2.0.0-p353/bin/ruby -S rspec ./spec/controllers/dashboards_controller_spec.rb ./spec/models/member_spec.rb ./spec/requests/members_spec.rb ./spec/routing/members_routing_spec.rb
[deprecated] I18n.enforce_available_locales will default to true in the future. If you really want to skip validation of your locale you can set I18n.enforce_available_locales = false to avoid this message.
 11/11 |============================================ 100 ============================================>| Time: 00:00:00 

Finished in 0.21233 seconds
11 examples, 0 failures

Randomized with seed 15954

/Users/josh/.rvm/rubies/ruby-2.0.0-p353/bin/ruby -S rspec ./spec/controllers/dashboards_controller_spec.rb ./spec/models/member_spec.rb ./spec/requests/members_spec.rb ./spec/routing/members_routing_spec.rb
[deprecated] I18n.enforce_available_locales will default to true in the future. If you really want to skip validation of your locale you can set I18n.enforce_available_locales = false to avoid this message.
 11/11 |============================================ 100 ============================================>| Time: 00:00:00 

Finished in 0.18831 seconds
11 examples, 0 failures

Randomized with seed 24248
```

When putting it only into the `:test` group, the specs are only executed once (as expected).

After some research I found that the `spec` rake task seems to be defined twice when `rspec-rails` is in both the `:test` and the `:development` groups (notice the `/` that separates the description of each task):

```
$ rake -T | grep spec
...
rake spec                               # Run all specs in spec directory (excluding plugin specs) / Run RSpec code examples
...
```

One of them is described with `Run all specs in spec directory (excluding plugin specs)`, one with `Run RSpec code examples`.

`Run RSpec code examples` seems to come from [`rspec/core/rake_task.rb`](https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/rake_task.rb#L72). `Run all specs in spec directory (excluding plugin specs)` seems to come from [`rspec/rails/tasks/rspec.rake`](https://github.com/rspec/rspec-rails/blob/master/lib/rspec/rails/tasks/rspec.rake#L12).

This looks definitely strange to me. So my question: why exactly should `rspec-rails` be also in the `:development` group (my specs seem to work perfectly when they're not)?

Thank you.

PS: I first opened a StackOverflow question about this issue, it's here: http://stackoverflow.com/questions/20872977/rails-and-rspec-rake-spec-runs-all-specs-twice

**Comment by lazylester** [comment]
I was referred here from issue #2169, I'm trying to understand how to invoke my test suite with `rake` (no args, so default task) with my own custom pattern match spec. What do I need to put in Rakefile to achieve this? Very confused and grateful for any suggestions, thanks.

**Comment by benoittgt** [comment]
Hello @lazylester 

You should look at:
- https://relishapp.com/rspec/rspec-core/v/3-8/docs/configuration/pattern
- https://relishapp.com/rspec/rspec-core/v/3-8/docs/configuration/exclude-pattern

Probably what your are looking for is close to:

```ruby
RSpec.configure do |config|
  config.pattern = "spec/models/*_spec.rb"
end
```

**Comment by lazylester** [comment]
thanks, @benoittgt the stack overflow question gave me the solution.
I'm not using the `rspec` command directly b/c the file pattern is a bit complex (I have many engines in the vendor/gems directory in addition to the top-level spec directory)
grateful for your assistance

**Comment by JonRowe** [comment]
You can define the pattern when defining the rake task.  See: http://rspec.info/documentation/3.8/rspec-core/RSpec/Core/RakeTask.html

```Ruby
RSpec::Core::RakeTask.new(:spec) do |task|
  task.pattern = "your_pattern"
end
```

**Comment by JonRowe** [comment]
In the case of your reproduction in#2169, you are getting two specs runs for the same reason as this issue, there are two spec tasks defined.

The TL;DR is define any overrides like this:
```Ruby
namespace :spec do
  RSpec::Core::RakeTask.new(:spec) do |task|
     task.pattern = "your_pattern"
  end
end
```

The longer explanation is that RSpec Rails automatically defines a task called "spec" within a namespace "spec" and then sets the default task for the namespace to be task, so `rake spec` is a shorthand for `rake spec:spec`. Just as `rake` is a shorthand for `rake spec` when you define `spec` to be the default task. So when you define another task called spec, there are now two tasks, `spec:spec` and `spec` triggered by `rake spec, and rake can't decide between them so runs both and if you followed that congratulations because I got confused just writing that 😂

**Comment by lazylester** [comment]
wow @JonRowe, this certainly is confusing! Thank you for taking the time to dig in and explain this.
